apps/kyb-app/src/components/organisms/UIRenderer/elements/JSONForm/components/DocumentField/DocumentField.tsx

```diff
@@ -121,20 +135,81 @@ export const DocumentField = (
           return;
         }
 
-        throw err;
+        throw error;
       } finally {
         toggleElementLoading();
       }
     },
-    [stateApi, options, definition, toggleElementLoading, sendEvent],
+    [stateApi, options, inputIndex, definition.valueDestination, toggleElementLoading],
+  );
+
+  const handleChange = useCallback(
+    (fileId: string) => {
+      //@ts-ignore
+      const destinationParser = new DocumentValueDestinationParser(definition.valueDestination);
+      const pathToDocumentsList = destinationParser.extractRootPath();
+      const pathToPage = destinationParser.extractPagePath();
+      const pathToFileId = destinationParser.extractFileIdPath();
+
+      const context = stateApi.getContext();
+      //@ts-ignore
+      const documents = (get(context, pathToDocumentsList) as Document[]) || [];
+
+      let document = documents.find(
+        document =>
+          document &&
+          //@ts-ignore
+          document.id === serializeDocumentId(definition.options.documentData.id, inputIndex),
```

Comment:
Is both sides being undefined an option?
```suggestion
          //@ts-ignore
          document?.id === serializeDocumentId(definition.options.documentData.id, inputIndex),
```